.github/workflows/upload-test-coverage.yml

```
@@ -0,0 +1,65 @@
+name: Upload test coverage report to GCS
+
+on:
+  schedule:
+    - cron: '0 0 1 * *' # At midnight on the 1st of every month
+  workflow_dispatch:
+
+permissions:
+  packages: read
+
+jobs:
+  collect-and-upload:
+    runs-on: ubuntu-latest
+
+    env:
+      BUCKET_NAME: dialtone-adoption-data
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set TIMESTAMP env var
+        run: echo "TIMESTAMP=$(date -u +'%Y-%m-%dT%H-%M')" >> $GITHUB_ENV
+
+      - name: Use Node.js 22
+        uses: actions/setup-node@v4
+        with:
+          node-version: 22
+
+      - name: Setup pnpm
+        uses: pnpm/action-setup@v3
+        with:
+          version: 8
+
+      - name: Authenticate to GitHub Packages
+        run: |
+          echo "//npm.pkg.github.com/:_authToken=${{ secrets.GITHUB_TOKEN }}" > ~/.npmrc
+          echo "@dialpad:registry=https://npm.pkg.github.com/" >> ~/.npmrc
+
+      - name: Install dependencies
+        run: pnpm install --frozen-lockfile
```

User our `setup-environment` reusable action to ease the maintainability of node and pnpm versions.

```suggestion
      - name: Setup Environment
        uses: ./.github/actions/setup-environment
```

---

Nice! Fixed in [38ecfe5](https://github.com/dialpad/dialtone/pull/836/commits/38ecfe5eac25deaab9c9c3ff74a8a6897b2de820)